var/spack/repos/builtin/packages/py-yaspin/package.py

```diff
@@ -0,0 +1,20 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class PyYaspin(PythonPackage):
+    """Yet Another Terminal Spinner"""
+
+    homepage = "https://github.com/pavdmyt/yaspin"
+    pypi     = "yaspin/yaspin-2.1.0.tar.gz"
+
+    version('2.1.0', sha256='c8d34eca9fda3f4dfbe59f57f3cf0f3641af3eefbf1544fbeb9b3bacf82c580a')
+
+    depends_on('python@3.6.2:3',       type=('build', 'run'))
+    depends_on('py-poetry-core@1:',    type='build')
+    depends_on('py-termcolor@1.1.0:1', type=('build', 'run'))
+    depends_on('py-dataclasses@0.8:0', type=('build', 'run'), when='^python@3.6')
```

Comment:
Poetry has a special syntax for this that behaves differently depending on whether the version starts with 0 or not (https://python-poetry.org/docs/dependency-specification/#caret-requirements). So I think this should actually be:
```suggestion
    depends_on('py-dataclasses@0.8', type=('build', 'run'), when='^python@3.6')
```